### ISSUE TITLE
Date Record not accesible in page extension

### ISSUE THREAD

**Issue by josecarlosherrero** [Microsoft/AL#4794]
**Bug**
When you publish an extension that accesses the virtual table Date returns that it is not accessible

**Code:**
Sample code:
```
pageextension 50102 GLAccountBalanceLinesExt extends "G/L Account Balance Lines"
{
    trigger OnAfterGetRecord();
    begin
        item.SetRange("Date Filter", Rec."Period Start", Rec."Period End");
        item.SetRange("Date Filter", "Period Start", "Period End");
    end;

    var
        item: Record Item;
}
```

Versión of Docker: 13.5.29614.0 (ES)
**Error:**
TestPageExtension03.al(5,38): error AL0118: The name 'Rec' does not exist in the current context
TestPageExtension03.al(5,58): error AL0118: The name 'Rec' does not exist in the current context
TestPageExtension03.al(6,38): error AL0118: The name 'Period Start' does not exist in the current context
TestPageExtension03.al(6,54): error AL0118: The name '"Period End"' does not exist in the current context

**It work in versión:**
Version in which it Works: 13.3.27312.0 (ES)

**Comment by thpeder** [comment]
Hi @josecarlosherrero, Could you please see if this reproduces on the latest developer preview.

**Comment by josecarlosherrero** [comment]
Yes, the same errors happen in the version:
**Version:** 14.0.29532.0 (US)
**Version:** 14.0.29532.0 (ES)

**Error:**
TestPageExtension03.al(5,38): error AL0118: The name 'Rec' does not exist in the current context
TestPageExtension03.al(5,58): error AL0118: The name 'Rec' does not exist in the current context
TestPageExtension03.al(6,38): error AL0118: The name 'Period Start' does not exist in the current context
TestPageExtension03.al(6,54): error AL0118: The name '"Period End"' does not exist in the current context

**Note:**
The error only appears when publishing, it does not give when compiling.

**Comment by josecarlosherrero** [comment]
**Solved un version**:  14.0.29537.31654 (ES)
